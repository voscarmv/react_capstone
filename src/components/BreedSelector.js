import React from 'react';
import propTypes from 'prop-types';
import { Dropdown, Button } from 'react-bootstrap';

const BreedSelector = ({ breedsData, handleSelectChange }) => (
  <Dropdown>
    <Dropdown.Toggle data-testid="breed_selector" variant="success" id="dropdown-basic">
      Choose
    </Dropdown.Toggle>
    <Dropdown.Menu>
      {breedsData.map(
        breed => (
          <Dropdown.Item key={breed.id}>
            <Button
              data-testid={breed.id}
              type="button"
              classList="dropdown-item"
              key={breed.id}
              onClick={handleSelectChange}
              value={breed.id}
            >
              { breed.name }
            </Button>
          </Dropdown.Item>
        ),
      )}
    </Dropdown.Menu>
  </Dropdown>
);

BreedSelector.propTypes = {
  breedsData: propTypes.arrayOf(propTypes.any).isRequired,
  handleSelectChange: propTypes.func.isRequired,
};

export default BreedSelector;
