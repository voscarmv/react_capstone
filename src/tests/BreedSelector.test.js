import React from 'react';
import TestRenderer from 'react-test-renderer';
import BreedSelector from '../components/BreedSelector';
import breeds from './BreedsResponse';

const BreedSelectorTest = TestRenderer.create(
  <BreedSelector breedsData={breeds} handleSelectChange={() => true} />,
);

it('Breed Selector renders correctly', () => {
  const tree = BreedSelectorTest.toJSON();
  expect(tree).toMatchSnapshot();
});
