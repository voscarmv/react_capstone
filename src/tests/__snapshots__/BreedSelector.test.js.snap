// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breed Selector renders correctly 1`] = `
<div
  className="dropdown"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-success"
    data-testid="breed_selector"
    disabled={false}
    id="dropdown-basic"
    onClick={[Function]}
    type="button"
  >
    Choose
  </button>
</div>
`;
